src/edu/stanford/dlss/wowza/SulWowza.java

```diff
@@ -15,73 +17,203 @@
         }
     }
 
-    // Invoked when an HTTP MPEGDash Streaming session is created;
-    // defined in IModuleOnHTTPMPEGDashStreamingSession module interfaces
+    /** Invoked when an HTTP MPEGDash Streaming session is created;
+     * defined in IModuleOnHTTPMPEGDashStreamingSession module interfaces */
     public void onHTTPMPEGDashStreamingSessionCreate(HTTPStreamerSessionMPEGDash httpSession)
     {
-        getLogger().info("Tommy: SulWowza onHTTPMPEGDashStreamingSessionCreate: " + httpSession.getSessionId());
+        getLogger().info(this.getClass().getSimpleName() + " onHTTPMPEGDashStreamingSessionCreate: " + httpSession.getStreamName());
         authorizeSession(httpSession);
     }
 
-    // Invoked when an HTTP Cupertino Streaming session is created;
-    // defined in IModuleOnHTTPCupertinoStreamingSession interface
+    /** Invoked when an HTTP Cupertino Streaming session is created  (for hls protocol (Safari, iphones));
+     * defined in IModuleOnHTTPCupertinoStreamingSession interface */
     public void onHTTPCupertinoStreamingSessionCreate(HTTPStreamerSessionCupertino httpSession)
     {
-        getLogger().info("Tommy: SulWowza onHTTPCupertinoStreamingSessionCreate: ");
+        getLogger().info(this.getClass().getSimpleName() + " onHTTPCupertinoStreamingSessionCreate: " + httpSession.getStreamName());
         authorizeSession(httpSession);
     }
 
     void authorizeSession(IHTTPStreamerSession httpSession)
     {
         String queryStr = httpSession.getQueryStr();
-        if (authorize(queryStr))
-            httpSession.acceptSession();
+        String stacksToken = getStacksToken(queryStr);
+        String userIp = httpSession.getIpAddress();
+        getLogger().debug(this.getClass().getSimpleName() + " userIp: " + userIp);
+        String streamName = httpSession.getStreamName();
+        getLogger().debug(this.getClass().getSimpleName() + " streamName: " + streamName);
+        if (validateStacksToken(stacksToken) && validateUserIp(userIp) && validateStreamName(streamName))
+        {
+            String druid = getDruid(streamName);
+            String filename = getFilename(streamName);
+
+            if (druid != null && filename != null && verifyStacksToken(stacksToken, druid, filename, userIp))
+                httpSession.acceptSession();
+            else
+                httpSession.rejectSession();
+        }
         else
             httpSession.rejectSession();
     }
 
-    boolean authorize(String queryStr)
+    /** Assumption: stacksToken, druid, userIp and filename are all reasonable values (non-null, not empty, etc.) */
+    boolean verifyStacksToken(String stacksToken, String druid, String filename, String userIp)
     {
-        String authToken = getAuthToken(queryStr);
-        return validateAuthToken(authToken);
+        String fullUrl = getVerifyStacksTokenUrl(stacksToken, druid, filename, userIp);
+        if (fullUrl != null)
+            return verifyStacksToken(fullUrl);
```

Comment:
Perhaps this is my Java ignorance, but it looks like the signature for `verifyStacksToken` takes token, druid, filename, and userIp strings.  In this logic branch it then calls itself with a fullUrl string.  Is that intentional?